src/main/kotlin/me/zeroeightsix/kami/feature/command/NbtCommand.kt

```diff
@@ -0,0 +1,221 @@
+package me.zeroeightsix.kami.feature.command
+
+import com.mojang.brigadier.CommandDispatcher
+import glm_.vec2.Vec2
+import glm_.vec4.Vec4
+import imgui.Col
+import imgui.TreeNodeFlag
+import imgui.api.childWindows
+import imgui.api.columns
+import imgui.api.contentRegion
+import imgui.api.cursorLayout
+import imgui.api.itemWidgetsUtilities
+import imgui.api.widgetsMain
+import imgui.api.widgetsText
+import imgui.api.widgetsTrees
+import imgui.api.windowsUtilities
+import imgui.dsl
+import me.zero.alpine.listener.Listener
+import me.zeroeightsix.kami.KamiMod
+import me.zeroeightsix.kami.event.TickEvent
+import me.zeroeightsix.kami.gui.InteractableScreen
+import me.zeroeightsix.kami.gui.KamiHud
+import me.zeroeightsix.kami.mc
+import me.zeroeightsix.kami.util.text
+import net.minecraft.client.util.math.MatrixStack
+import net.minecraft.command.CommandSource
+import net.minecraft.nbt.AbstractListTag
+import net.minecraft.nbt.AbstractNumberTag
+import net.minecraft.nbt.CompoundTag
+import net.minecraft.nbt.ListTag
+import net.minecraft.nbt.StringTag
+import net.minecraft.nbt.Tag
+import net.minecraft.text.MutableText
+import net.minecraft.text.Text
+import net.minecraft.util.Formatting
+import net.minecraft.util.hit.HitResult
+import net.minecraft.util.math.BlockPos
+
+object NbtCommand : Command() {
+    var screen = NbtScreen(CompoundTag())
+    var open = true
+
+    // this is required, as the chat closes all screens when a command is entered,
+    // so the screen needs to be opened a tick later
+    var opener: Listener<TickEvent.InGame>? = null
+
+    init {
+        opener = Listener({
+            mc.openScreen(screen)
+            KamiMod.EVENT_BUS.unsubscribe(opener)
+        })
+    }
+
+    override fun register(dispatcher: CommandDispatcher<CommandSource>) {
+        dispatcher register rootLiteral("nbt") {
+            literal("look") {
+                does {
+                    val target = mc.crosshairTarget
+                    val tag = when (target?.type) {
+                        HitResult.Type.BLOCK -> {
+                            val entity = mc.world?.getBlockEntity(BlockPos(target.pos))
+                            if (entity == null) {
+                                Error.NO_BLOCK_ENTITY.send(it.source)
```

Comment:
Errors in brigadier are supposed to be thrown exceptions. Brigadier will handle them differently and show the user a formatted error.

Take a look at `PeekCommand`'s `FAILED_EXCEPTION` and when it is thrown.